### DISCUSSION TITLE
dblclick event handlers registered against the wavesurfer object vs the drawer

### DISCUSSION THREAD

**Discussion by rewbs** [katspaugh/wavesurfer.js#2759] [General]
Hello!

There's an issue here about the `dblclick` event not invoking registered handlers: https://github.com/wavesurfer-js/wavesurfer.js/issues/2655 . The reporter closed it after identifying that you can work around the issue by registering the handler on the drawer object instead of the wavesurfer object.

I ran into this too, and while I'm happy to have a workaround, it's not entirely clear to me that it's intentional that we have to register the event handler on the drawer. Unless I'm missing something, the `dblclick` event is documented in the same way as the others events, but fires differently. Compare for example with the `scroll` event, which invokes handlers both on the Wavesurfer object and on the drawer. To me it intuitively "feels" like the event handlers on the drawer are more related to internal implementation, and as a consumer of the library, I should be registering handlers on the Wavesurfer object only.

Is there a small bug here, or is my intuition wrong? :)

Keep up the great work and many thanks for the awesome library!

**Comment by katspaugh** [comment]
Good catch, I agree 100% it should be on the wavesurfer object, especially as it's documented as such. Please feel free to make a PR, it should be straightforward.
